File: app/gui/MyInputStream.js
var JsonInputStream = require('./JsonInputStream');

var MyInputStream = function(input, inputHandler) {
    this.input = new JsonInputStream(input, function(jsonData) {
        var type = jsonData.type;
        var data = jsonData;
        delete data.type;
        inputHandler(type, data);
    });
}

module.exports = MyInputStream;

File: app/controller/MyInputStream.rb
require './JsonInputStream'

class MyInputStream
    def initialize input
        @jsonInputStream = JsonInputStream.new input do |jsonData|
            type = jsonData['type']
            data = jsonData
            data.delete 'type'
            yield type, data
        end
    end

    def stopReading
        @jsonInputStream.stopReading
    end
end
